servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -4695,6 +4695,24 @@ void processUserAuthorityRestrictions() {
         }
     }
 
+    Map<String, List<String>> applyTemplatesForListOfDomains(Map<String, Integer> templateDetails) {
+        final String caller = "applyTemplatesForListOfDomains";
+        final String auditRef = "AutoApplyTemplate";
+        Map<String, List<String>> domainTemplateListMap;
+        DomainTemplate domainTemplate = new DomainTemplate();
+        try (ObjectStoreConnection con = store.getConnection(true, false)) {
+             domainTemplateListMap = con.getDomainFromTemplateName(templateDetails);
+        }
+
+        for (String domainName : domainTemplateListMap.keySet()) {
+            domainTemplate.setTemplateNames(domainTemplateListMap.get(domainName));
+            //Passing null context since it is an internal call during app start up
+            //executePutDomainTemplate can bulk apply templates given a domain hence sending domainName and templatelist
+            this.executePutDomainTemplate(null, domainName, domainTemplate, auditRef, caller);
```

Comment:
this call should be protected such that if any exception is thrown, we don't stop the process but rather just swallow it and let it continue with the rest of domains.

Reply:
yep handled exception case to skip failures and continue.. thank you.